france_connect/clients.py

```diff
@@ -0,0 +1,192 @@
+        self.fc_token_url = fc_token_url
+        self.fc_jwks_url = fc_jwks_url
+        self.fc_userinfo_url = fc_userinfo_url
+        self.fc_logout_url = fc_logout_url
+        self.timeout = timeout
+        self.verify_ssl = verify_ssl
+        self.allow_redirects = allow_redirects
+
+    def generate_nonce(self) -> str:
+        """Generate a random nonce in hexadecimal string (64 characters).."""
+        return secrets.token_hex(48)
+
+    def generate_state(self) -> str:
+        """Generate a random state in hexadecimal string (64 characters)."""
+        return secrets.token_hex(48)
+
+    def get_authentication_url(
+        self, acr_values: List[ACRValues] = None, nonce: str = None, state: str = None
+    ) -> Tuple[str, str, str]:
+        """Return the authentication URL, nonce and state."""
+        acr_values = acr_values or [ACRValues.EIDAS1]
+        nonce = nonce or self.generate_nonce()
+        state = state or self.generate_state()
+        params = urllib.parse.urlencode(
+            {
+                "client_id": self.client_id,
+                "response_type": "code",
+                "scope": " ".join(self.scopes),
+                "acr_values": " ".join(acr_values),
+                "nonce": nonce,
+                "state": state,
+                "redirect_uri": self.post_login_callback_url,
+            }
+        )
+        url = urllib.parse.urljoin(self.fc_base_url, self.fc_authorization_url)
+        return f"{url}?{params}", nonce, state
+
+    def get_configuration(self) -> dict:
+        """Return the OpenID configuration."""
+        url = urllib.parse.urljoin(self.fc_base_url, "/api/v2/.well-known/openid-configuration")
+        try:
+            response = requests.get(
+                url, timeout=self.timeout, verify=self.verify_ssl, allow_redirects=self.allow_redirects
+            )
+            return response.json()
+        except requests.exceptions.RequestException:
+            msg = f"FranceConnect - Failed to retrieve configuration"
+            raise FranceConnectConfigurationError(response.status_code, response.text, msg)
+
+    def get_access_token(self, code: str) -> dict:
+        """Exchange the authorization code for an access token."""
+        token_url = urllib.parse.urljoin(self.fc_base_url, self.fc_token_url)
+        data = {
+            "grant_type": "authorization_code",
+            "code": code,
+            "redirect_uri": self.post_login_callback_url,
+            "client_id": self.client_id,
+            "client_secret": self.client_secret,
+        }
+        try:
+            response = requests.post(
+                token_url,
+                data=data,
+                headers={"Accept": "application/json"},
+                timeout=self.timeout,
+                verify=self.verify_ssl,
+                allow_redirects=self.allow_redirects,
+            )
+            response.raise_for_status()
+            return response.json()
```

Comment:
You should probably call `self.verify_access_token()` so the package's user don't have to do it himself.